src/uu/cp/src/cp.rs

```diff
@@ -1555,15 +1612,61 @@ fn copy_on_write_linux(
                 Ok(())
             }
         },
-        ReflinkMode::Auto => unsafe {
+        (ReflinkMode::Always, SparseMode::Always) | (ReflinkMode::Always, SparseMode::Never) => {
+            Err("`--reflink=always` can be used only with --sparse=auto".into())
+        }
+        (_, SparseMode::Always) => unsafe {
+            let size: usize = src_file.metadata()?.size().try_into().unwrap();
+            if libc::ftruncate(dst_file.as_raw_fd(), i64::try_from(size).unwrap()) < 0 {
```

Comment:
I guess you saw the "mismatched types" error

Reply:
I did this for a similar issue:
https://github.com/uutils/coreutils/pull/3614/files#diff-fed63f74214c092a801901cdce011f1941f6378a84d9c7947b5bd02adce2de43R227-R235

Reply:
Thanks for the pointer, I think `.try_into().unwrap()` is sufficient (at least I manage to build locally with `CC=clang cargo test --target=i686-unknown-linux-gnu`)